### ISSUE TITLE
Remove Numpy*Domains?

### ISSUE THREAD

**Issue by jnothman** [numpy/numpydoc#138]
From #134:

Some tracing tells me that wrap_mangling_directives results in NumpyPythonDomain's directives being used to handle py:class, py:function, etc. I find it hard to believe that that's the intention.

Do these domains remain relevant? I can't see any reference to them in the numpy or scipy repositories currently. I see that @pv [commented](https://github.com/numpy/numpy/pull/379#issuecomment-7638375) on their likely irrelevance in 2012.

**Comment by pv** [comment]
Yes, probably the `self.directives` dictionary should be copied.

If you want to write documentation in numpydoc syntax using the sphinx `.. function::` way instead of getting them from docstrings via autodoc, then a separate domain is the solution. I don't know if they are used in the wild, but on the other hand supporting them is not that expensive.

**Comment by jnothman** [comment]
Do you think that's what's happening? It's overwriting the base directives?

**Comment by jnothman** [comment]
If there were tests for this, I'd be better assured when trying that change... :\
